internal/controller/warehouses/warehouses.go

```diff
@@ -191,38 +194,169 @@ func (r *reconciler) syncWarehouse(
 	ctx context.Context,
 	warehouse *kargoapi.Warehouse,
 ) (kargoapi.WarehouseStatus, error) {
+	logger := logging.LoggerFromContext(ctx)
+
 	status := *warehouse.Status.DeepCopy()
-	status.ObservedGeneration = warehouse.Generation
-	status.Message = "" // Clear any previous error
 
 	// Record the current refresh token as having been handled.
 	if token, ok := kargoapi.RefreshAnnotationValue(warehouse.GetAnnotations()); ok {
 		status.LastHandledRefresh = token
 	}
 
-	logger := logging.LoggerFromContext(ctx)
+	// Clear previous error message.
+	// TODO(hidde): Remove this once we have removed the deprecated field.
+	status.Message = "" // nolint:staticcheck
 
 	// Discover the latest artifacts.
-	discoveredArtifacts, err := r.discoverArtifactsFn(ctx, warehouse)
-	if err != nil {
-		return status, fmt.Errorf("error discovering artifacts: %w", err)
+	if shouldDiscoverArtifacts(warehouse, status.LastHandledRefresh) {
+		// As this is a long-running operation, we need to update the status
+		// conditions to reflect that we are currently reconciling.
+		conditions.Set(
+			&status,
+			&metav1.Condition{
+				Type:   kargoapi.ConditionTypeReconciling,
+				Status: metav1.ConditionTrue,
+				Reason: "ScheduledDiscovery",
+				Message: fmt.Sprintf(
+					"Discovering artifacts for %d subscriptions",
+					len(warehouse.Spec.Subscriptions),
+				),
+				ObservedGeneration: warehouse.GetGeneration(),
+			},
+			&metav1.Condition{
+				Type:               kargoapi.ConditionTypeReady,
+				Status:             metav1.ConditionFalse,
+				Reason:             "AwaitingDiscovery",
```

Comment:
`AwaitingDiscovery` might be misleading. I think you said this is true when waiting for discovery to complete, but `AwaitingDiscovery` sounds to me like waiting for discovery to _begin_.

Any thoughts on how to clarify?

Reply:
`AwaitingDiscoveryResults` or `WaitingForDiscovery` then maybe?

Reply:
`WaitingForDiscovery` feels the same as `AwaitingDiscovery`.

I am fine with `AwaitingDiscoveryResults`... or maybe `DiscoveryInProgress`? Do you feel as if the latter would accurately convey the condition you are trying to convey?